### ISSUE TITLE
sanity color-input won't take a custom icon

### ISSUE THREAD

**Issue by ryan-reach** [sanity-io/sanity#1917]
**Describe the bug**

Cannot set blockEditor.icon for 'color' type from @sanity/color-input

**To Reproduce**

1. Create a fresh project with sanity init and choose blog
2. sanity install @sanity/color-input
3. add color type to blockContent 'block' type in marks.annotations:
```
          {
            title: "Color",
            type: "color",
            blockEditor: {
              icon: () => "🎨",
            },
          },
```
4. open studio and select a new Post type
5. check Body section PortableText editor
6. default 'S' icon still showing
7. add blockEditor.icon to URL object type named 'link' in example project
8. you can see the custom icon works for 'object' type, but not 'color'

**Expected behavior**

I expected sanity's color-input type to show custom icons just like all other types

**Screenshots**

custom icon on 'color' type:
<img width="619" alt="Screen Shot 2020-06-08 at 1 04 03 PM" src="https://user-images.githubusercontent.com/60242214/84070047-bd9c3980-a988-11ea-8c43-dface62f3e90.png">

custom icon on link with 'object' type:
<img width="621" alt="Screen Shot 2020-06-08 at 1 03 46 PM" src="https://user-images.githubusercontent.com/60242214/84069987-a5c4b580-a988-11ea-82f1-cbd76903e355.png">

**Which versions of Sanity are you using?**

@sanity/cli             1.149.13 (up to date)
@sanity/base            1.149.15 (up to date)
@sanity/color-input      1.149.7 (up to date)
@sanity/components      1.149.15 (up to date)
@sanity/core            1.149.13 (up to date)
@sanity/default-layout  1.149.15 (up to date)
@sanity/default-login   1.149.11 (up to date)
@sanity/desk-tool       1.149.15 (up to date)
@sanity/vision           1.149.0 (up to date)

**What operating system are you using?**

macOs 10.15.4

**Which versions of Node.js / npm are you running?**

6.14.4
v10.21.0

**Additional context**

code from blockContent.js with blockEditor.icon on both 'color' type and link with 'object' type. icon only shows on link...

```
import React from "react";

export default {
  title: "Block Content",
  name: "blockContent",
  type: "array",
  of: [
    {
      title: "Block",
      type: "block",

      styles: [
        { title: "Normal", value: "normal" },
        { title: "H1", value: "h1" },
        { title: "H2", value: "h2" },
        { title: "H3", value: "h3" },
        { title: "H4", value: "h4" },
        { title: "Quote", value: "blockquote" },
      ],
      lists: [{ title: "Bullet", value: "bullet" }],
      marks: {
        decorators: [
          { title: "Strong", value: "strong" },
          { title: "Emphasis", value: "em" },
        ],
        annotations: [
          {
            title: "URL",
            name: "link",
            type: "object",
            fields: [
              {
                title: "URL",
                name: "href",
                type: "url",
              },
            ],
            blockEditor: {
              icon: () => "🎨",
            },
          },
          {
            title: "Color",
            name: "color",
            type: "color",
            blockEditor: {
              icon: () => "🎨",
            },
          },
        ],
      },
    },
    {
      type: "image",
      options: { hotspot: true },
    },
  ],
};
```

**Security issue?**
no

**Comment by ryan-reach** [comment]
after checking out the code in @sanity/color-input/lib/schemas/color.js, it doesn't look like the `blockEditor` is part of the schema. When I add it, it works.

I inspected the AnnotationButtons with react-devtools, and saw that for both the link and color buttons, their `blockEditor` props are undefined (despite having defined it in the schema for the color input). I also noticed that these buttons have an `icon` prop, outside of the `blockEditor` object. When using the `icon` prop that sits at the same level as `type`, `name`, and `title`, sure enough my custom icon shows up.

I'm guessing the `icon` prop is there at the root for backwards compatibility. But it seems like blockEditor should not be undefined in the first place.

<img width="774" alt="Screen Shot 2020-06-08 at 3 43 06 PM" src="https://user-images.githubusercontent.com/60242214/84084216-46729f80-a9a0-11ea-8c0d-9ffcb10e20f0.png">

**Comment by vicbergquist** [comment]
Thank you for reporting, this looks like a regression. We'll look into it! I believe this is also related 👍 https://github.com/sanity-io/sanity/issues/1536

**Comment by mikebm** [comment]
Same issue - Was hoping to set a custom block editor to render the font color in the portable text view, but it won't accept a block editor

**Comment by arthurdenner** [comment]
Just FYI, there is a proposed fix for this in #2273 waiting for review.
